plugins/modules/rudder_node_settings.py

```diff
@@ -257,38 +258,36 @@ def _send_request(
             rudder_url=self.rudder_url, path=path
         )
 
-        if method == 'POST':
-            resp = requests.post(
-                url=full_url,
-                headers=self.headers,
-                data=s_data,
-                params=params,
-                verify=verify,
-            )
+        resp, info = fetch_url(
+            method=method,
+            url=full_url,
+            data=s_data,
+            headers=self.headers,
+            verify=verify,
+        )
 
-        elif method == 'GET':
-            resp = requests.get(
-                url=full_url,
-                headers=self.headers,
-                params=params,
-                verify=verify,
+        status_code = info['status']
+        if status_code == 404:
+            return None
```

Comment:
? I believe we should expose such error in the output

Reply:
Fixup !